control_plane/src/endpoint.rs

```diff
@@ -630,9 +644,35 @@ impl Endpoint {
             pgbouncer_settings: None,
             shard_stripe_size: Some(shard_stripe_size),
             local_proxy_config: None,
-            reconfigure_concurrency: 1,
+            reconfigure_concurrency: self.reconfigure_concurrency,
             drop_subscriptions_before_start: self.drop_subscriptions_before_start,
         };
+
+        // this strange code is needed to support respec() in tests
+        if self.cluster.is_some() {
```

Comment:
Comment for the future. I don't really like this endpoint.rs code, it's a mess. We have spec.json file and endpoint.json file, sometimes they are used interchangeably, sometimes not

This endpoint.json is a subset of the actual spec, so it doesn't allow us to fully control all attributes during tests

I'm not sure what's the best design here, maybe have some generic `spec_overrides` in the `endpoint.json`, which is then merged with the spec generated by `endpoint.rs` and overrides any intersecting items. Similar to how I hacked it in `respec_deep` fixture (which I do not like either)

Here we add even more crutches around it, so it complicates it further

Reply:
+1 
I also don't like this code.

Should we just move to e2e tests for everything that relies on spec/ respec logic and drop endpoint.rs and local cplane completely?

Reply:
With e2e we are limited with what external APIs in console/cplane exist, they are also more heavy-weighted. Here you can pass 10 DBs via spec, in cplane you would need to call API 10 times and schedule 10 operations. Same for everything. I'd still have some good fixture here